### PR TITLE
ONEUP-2217: reverted changes for redirect exception

### PR DESCRIPTION
**Pull request by Normal**

### CODE REVIEW

File: core/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala [hunk 1]
```diff
@@ -154,13 +154,13 @@ abstract class AbstractService[T](ws: WSClient,
     metrics.defaultRegistry.timer(MetricRegistry.name(s"service.$name", "requestTimer")).time()
   }
 
-  private[contentapi] def processResponse(response: WSResponse, uri: String): T = response.status match {
+  private[contentapi] def processResponse(response: WSResponse, request: WSRequest): T = response.status match {
     case Status.OK | Status.CREATED => processResult(response)
-    case status if (300 until 400).contains(status) => throw HttpRedirectException(uri, response.statusText)
-    case Status.UNAUTHORIZED  => throw HttpServerErrorException(Status.NETWORK_AUTHENTICATION_REQUIRED, response.statusText, uri)
+    case status if (300 until 400).contains(status) => throw HttpRedirectException(request.url, response.statusText)
```
Comment (Normal) [inline core/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala:159]: I think we shouldn't touch `HttpRedirectException` in previous commit, funkotron has some business with it so I lower url -> uri change for error case only.

File: core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala [hunk 2]
```diff
@@ -5,25 +5,25 @@ import play.api.http.Status
...
 
-    def getUrl: String = uri
+    def getUrl: String = url
 
     override def toString: String = super.toString
   }
 
-  case class HttpRedirectException(uri: String, statusPhrase: String)
-    extends HttpStatusCodeException(MOVED_PERMANENTLY, statusPhrase, uri)
+  case class HttpRedirectException(url: String, statusPhrase: String)
```
Comment (yolch-yolchyan) [inline core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala:20]: Should it be a `uri`?

Comment (Normal) [reply core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala:20]: In my mind it should be `uri` for `HttpClientErrorException` and `HttpServerErrorException` only.

Comment (yolch-yolchyan) [reply core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala:20]: If this is a link for a specific page then I think it should be `uri`. But, I'm ok with URL too

**Review by yolch-yolchyan — APPROVED** [review]

**Review by catalin-cretu — APPROVED** [review]

File: core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala (hunk 2 above)
Comment (catalin-cretu) [reply core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala:20]: URI and URL has confusing definitions so we shouldn't have to even choose between the 2.
But I guess that's something coming from play which is sad.

Comment (Normal) [reply core/src/main/scala/de/welt/contentapi/core/client/services/exceptions.scala:20]: @yolch-yolchyan jfyi it was always `url` here, we renamed it to `uri` in the last commit which we want to partially revert now. I agree with you that it makes sense to define `uri` for some of them but I don't like the idea that they all extend the same base abstract class `HttpStatusCodeException` (doesn't look like a scala way to me).

I would say we need to refactor that piece (together with `AbstractService`) later.

**Pull request merged**